modules/pubwiseAnalyticsAdapter.js

```diff
@@ -54,72 +77,253 @@ function enrichWithUTM(dataBag) {
+  }
+  if (typeof modified.bidsReceived !== 'undefined') {
+    delete modified.bidsReceived;
+  }
+  if (typeof modified.config !== 'undefined') {
+    delete modified.config;
+  }
+  if (typeof modified.noBids !== 'undefined') {
+    delete modified.noBids;
+  }
+  if (typeof modified.winningBids !== 'undefined') {
+    delete modified.winningBids;
   }
 
-  ajax(configOptions.endpoint, (result) => utils.logInfo(`${analyticsName}Result`, result), JSON.stringify(dataBag));
+  return modified;
 }
 
-let pubwiseAnalytics = Object.assign(adapter(
-  {
-    defaultUrl,
-    analyticsType
-  }),
-{
+let pubwiseAnalytics = Object.assign(adapter({defaultUrl, analyticsType}), {
   // Override AnalyticsAdapter functions by supplying custom methods
   track({eventType, args}) {
-    sendEvent(eventType, args);
+    this.handleEvent(eventType, args);
   }
 });
 
+pubwiseAnalytics.handleEvent = function(eventType, data) {
+  // we log most events, but some are information
+  if (isIngestedEvent(eventType)) {
+    pwInfo(`Emitting Event ${eventType} ${pwAnalyticsEnabled}`, data);
+
+    // record metadata
+    metaData = {
+      target_site: configOptions.site,
+      debug: configOptions.debug ? 1 : 0,
+    };
+    metaData = enrichWithSessionInfo(metaData);
+    metaData = enrichWithMetrics(metaData);
+    metaData = enrichWithUTM(metaData);
+    metaData = enrichWithCustomSegments(metaData);
+
+    // add data on init to the metadata container
+    if (eventType === CONSTANTS.EVENTS.AUCTION_INIT) {
+      data = filterAuctionInit(data);
+    } else if (eventType === CONSTANTS.EVENTS.BID_RESPONSE) {
+      data = filterBidResponse(data);
+    }
+
+    // add all ingested events
+    pwEvents.push({
+      eventType: eventType,
+      args: data
+    });
+  } else {
+    pwInfo(`Skipping Event ${eventType} ${pwAnalyticsEnabled}`, data);
+  }
+
+  // once the auction ends, or the event is a bid won send events
+  if (eventType === CONSTANTS.EVENTS.AUCTION_END || eventType === CONSTANTS.EVENTS.BID_WON) {
+    flushEvents();
+  }
+}
+
+pubwiseAnalytics.storeSessionID = function (userSessID) {
+  storage.setDataInLocalStorage(localStorageSessName(), userSessID);
+  pwInfo(`New Session Generated`, userSessID);
+};
+
+// ensure a session exists, if not make one, always store it
+pubwiseAnalytics.ensureSession = function () {
+  if (sessionExpired() === true || userSessionID() === null || userSessionID() === '') {
+    let generatedId = utils.generateUUID();
+    expireUtmData();
+    this.storeSessionID(generatedId);
+    sessionData.sessionId = generatedId;
+  }
+  // eslint-disable-next-line
+  // console.log('ensured session');
+  extendUserSessionTimeout();
+};
+
 pubwiseAnalytics.adapterEnableAnalytics = pubwiseAnalytics.enableAnalytics;
 
 pubwiseAnalytics.enableAnalytics = function (config) {
-  if (config.options.debug === undefined) {
-    config.options.debug = utils.debugTurnedOn();
-  }
   configOptions = config.options;
```

Comment:
So, I have probably found an issue here. Based on the docs you've written at the top of this file, I added these lines to my test page: 
```
pbjs.enableAnalytics({
          provider: "pubwise",
          options: {
            site: "test-test-test-test",
          },
        });
```
when the execution gets here, `configOptions.endpoint` was set to `defaultUrl`, but that gets removed by this assignment: 

```
configOptions = config.options;
```

As a result, i see a bunch of errors in console: 

```
POST http://localhost:9999/integrationExamples/gpt/undefined 405 (Method Not Allowed)
(anonymous) @ ajax.js:91
flushEvents @ pubwiseAnalyticsAdapter.js:171
...
Prebid ERROR: xhr error null Method Not Allowed
```
Note the  **undefined** at the end of the url. 

So, `configOptions.endpoint` does not exist anymore. 

I think you wanted to extend configOptions with the data coming from function arguments: 
```
configOptions = {
    ...configOptions, 
    ...config.options  
}
```
(This can be also achieved by using Object.assign)

Can you have a look into this?

Reply:
@musikele this is resolved now